src/tasks/actions/thunks.ts

```diff
@@ -193,6 +193,39 @@ export const updateTaskName = (name: string, taskID: string) => async (
   }
 }
 
+type UpdateTaskParams = {
+  script: string
+  preamble: string
+  task: Task
+  interval: string
+}
+
+export const updateTask = ({
+  script,
+  interval,
+  preamble,
+  task,
+}: UpdateTaskParams) => async (dispatch: Dispatch<Action>) => {
+  try {
+    const fluxScript = await insertPreambleInScript(script, preamble)
```

Comment:
did you get clarification from russ on the concept of system variables in task queries? they still dont make any sense to me

Reply:
I didn't, let me run that by him right now.